File: src/crud_nodejs_express_ejs/infrastructure/repositories.ts
import type { CrudNodejsExpressEjs } from '../domain/entities';
import type { CrudNodejsExpressEjsRepository } from '../domain/repositories';

export class InMemoryCrudNodejsExpressEjsRepository implements CrudNodejsExpressEjsRepository {
  private readonly items = new Map<string, CrudNodejsExpressEjs>();

  async save(entity: CrudNodejsExpressEjs): Promise<CrudNodejsExpressEjs> {
    this.items.set(entity.id, entity);
    return entity;
  }
}

File: src/crud_feature/infrastructure/repositories.py
from __future__ import annotations

from ..domain.entities import CrudFeature
from ..domain.repositories import CrudFeatureRepository


class InMemoryCrudFeatureRepository(CrudFeatureRepository):
    def __init__(self) -> None:
        self._items: dict[str, CrudFeature] = {}

    def save(self, entity: CrudFeature) -> CrudFeature:
        self._items[entity.id] = entity
        return entity
